### DISCUSSION TITLE
Is Python SDK of LanceDB Asynchronous?

### DISCUSSION THREAD

**Discussion by tjtanaa** [lancedb/lancedb#1067] [Q&A] [answered]
Does anyone know if Python SDK of LanceDB Asynchronous?
I saw the Javascript SDK example is async.

**Comment by wjones127** [comment, marked as answer]
Currently, it is synchronous (not async). However, the method calls do not hold the GIL. So in async contexts, you should use something like [run_in_executor](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor) to run the calls in the background and await them on your main thread.

However, in the next few weeks we will release a new version of Python SDK with an async API. So soon that will all be unnecessary.

**Reply by tjtanaa** [reply]
Thank you. Looking forward to the feature.
